src/widget.ts

```diff
@@ -405,6 +409,22 @@ export class CytoscapeView extends DOMWidgetView {
         elements: this.model.get('graph').converts_dict(),
       });
 
+      const monitored = this.model.get('monitored');
+      for (const widgtype in monitored) {
+        if (Object.prototype.hasOwnProperty.call(monitored, widgtype)) {
+          for (let i = 0; i < monitored[widgtype].length; i++) {
+            const evnttype = monitored[widgtype][i];
+            this.cytoscape_obj.on(evnttype, widgtype, (e: any) => {
+              this.send({
+                event: evnttype,
+                widget: widgtype,
+                data: e.target.data(),
```

Comment:
```suggestion
                data: e.target,
```
or maybe even just `e`. Is there a reason to limit the information that gets transmitted back to python? For example I tend to store some information in the `classes` attribute which would be removed from what is returned here.

Reply:
Good point; I'll switch it to `e.target`. Is `e` itself serializable, though?

Reply:
My guess would be yes, but just trying is probably the best way to find out.

Reply:
I tried serializing and sending back the full target widget, but you can't do it because it's a cyclic structure. I think this is fine anyway since in principle even if we could serialize it now we'd have no guarantee that it's non-cyclic in the future. We also in principle don't know the limits on JSON representation size of objects outside of our control/user control. Also, since the objects are set from the kernel, I think it's reasonable to assume the data is available in the kernel anyway. If a unique ID was always required from the client, I would even go so far as to recommend that that be the only data returned, but since it's possible to load arbitrary data dictionaries, I think it's adequate to require the user to make sure they can use the returned dict to extract some unique identifier.